### ISSUE TITLE
QUZ: Nutzerverwaltung

### ISSUE THREAD

**Issue by JulianeOrt** [imbitdhbwma/brillianplatform#66]
<img width="1440" alt="Screenshot 2019-11-14 at 11 57 49" src="https://user-images.githubusercontent.com/36894965/68851987-3da23f80-06d7-11ea-8a74-4e9ae8f32cf0.png">

Im Quiz sollte man sich anmelden können. Eine Neu Registrierung ist aber nicht möglich. Könntet ihr euch das anschauen?

**Comment by JulianeOrt** [comment]
<img width="1440" alt="Screenshot 2019-11-14 at 12 09 00" src="https://user-images.githubusercontent.com/36894965/68852122-8fe36080-06d7-11ea-8e3e-424fee36cc68.png">

hier der Screenshot zur Fehlermeldung.

**Comment by Lara-u** [comment]
Kann man ein Konto erstellen? Funktioniert das ?

**Comment by Lara-u** [comment]
<img width="511" alt="Bildschirmfoto 2019-11-15 um 09 34 58" src="https://user-images.githubusercontent.com/56717795/68928820-4b190180-078b-11ea-9b4a-400fc5eb5317.png">

Ok, die Funktion schein momentan generell noch nicht verfügbar zu sein! Wir schauen uns das an!

**Comment by Myprr** [comment]
Update:
Für die Benutzerverwaltung ist eine Datenbank auf dem Server vorausgesetzt.

Auf dem Q Server ist bereits eine Datenbank für BrillianQuiz angelegt. Diese wird noch geprüft. (Auf dem P Server nicht)
@pnt-linii prüft ob die DB auf QServer mit dem Test Quiz verbunden ist.

Ansonsten wird eine neue DB auf dem QServer angelegt und getestet. Danach wirds im PServer implementiert.

**Comment by JulianeOrt** [comment]
Admin Seite wie ICM/CRM integrieren, Seite für Studiengangsleitung wo er die Kurse und pro Kurs verschiedene Dozenten zu Vorlesungsquiz hinzufügen kann als Matrix (Zeilen sind die Kurse und Spalten die Vorlesung, pro Matrixpunkt kann er dann einen Dozenten auswählen), Dozenten bekommen die Möglichkeit Prüfungsquizze für ihre Studenten für ca. 72h freizuschalten und dieses Prüfungsquiz auch zu beenden. 
Login für alle über Krakenkopf unten rechts im QUZ. Sobald die Studenten eingeloggt sind, sehen sie nur die für die Prüfungen relevanten Quizze und haben bestimmte Zeit diese zu bearbeiten, können aber nur einmal auf “zur Bewertung abgeben” drücken, dann ist die Abgabe auch final? Die Admin/Studiengangsleiter/Dozenten loggen sich auch über den Krakenkopf ein. Für sie muss die Anmutung aber erst erstellt werden. Die Seite soll zudem mit einer “Wasserspritzpistole a la James Bond 007” ausgestattet werden, der Hintergrund im Admin Bereich passend dazu.

- Das Design Team kümmert sich um die visuelle Oberfläche und UX.
- Das Technik Team sollte sich bereits mit den Daten auseinandersetzen, welche Datenbank kann genutzt werden,... (bitte bei Fragen dazu direkt mit PO abstimmen bzw. hat Tobi dazu schon einen Einblick bekommen)

**Comment by JulianeOrt** [comment]
ACHTUNG: HOHE PRIORITÄT & sollte Ende des 3. Sprints möglichst abgeschlossen sein!

**Comment by pnt-linii** [comment]
Verbindung mit Datenbank auf dem QServer besteht nach meinem Testergebnis. Login-Versuch mit public@brillianquiz.com war auch erfolgreich.
@Myprr

**Comment by pnt-linii** [comment]
TODOs: 
- "Passwort Vergessen-Seite" und "Registrierung-Seite" bedürfen Anpassungen bezüglich Designs. Bitte jemand sich darum kümmere.

**Comment by pnt-linii** [comment]
> @Myprr: Verbindung mit Datenbank auf dem QServer besteht nach meinem Testergebnis. Login-Versuch mit [public@brillianquiz.com](mailto:public@brillianquiz.com) war auch erfolgreich.
> 
> !! Es wurde keine E-Mail rausgeschickt für Registrierung. Ich werde mich darum kümmern.

Update: E-Mail Server funktioniert in der Tat aber diese Funktion wird momentan im Test-Server deaktiviert im Kontext vom Issue #53

**Comment by Myprr** [comment]
> 
> 
> TODOs:
> 
>     * "Passwort Vergessen-Seite" und "Registrierung-Seite" bedürfen Anpassungen bezüglich Designs. Bitte jemand sich darum kümmere.
> 
>     * Fix Registrationsbestätigungslink (Link in Bestätigungsemail führt zur 404)

@Lara-u erstelle bitte separate tasks für die oberen Aufgaben und verlinke mit diesem Task.

@pnt-linii Task in diesem Issue ist also die Konfiguration von Test auf Prod zu migrieren und mit unserer aktiver Mailfunktion auszutesten. (?)

**Comment by pnt-linii** [comment]
> > TODOs:
> > ```
> > * "Passwort Vergessen-Seite" und "Registrierung-Seite" bedürfen Anpassungen bezüglich Designs. Bitte jemand sich darum kümmere.
> > 
> > * Fix Registrationsbestätigungslink (Link in Bestätigungsemail führt zur 404)
> > ```
> 
> @Lara-u erstelle bitte separate tasks für die oberen Aufgaben und verlinke mit diesem Task.
> 
> @pnt-linii Task in diesem Issue ist also die Konfiguration von Test auf Prod zu migrieren und mit unserer aktiver Mailfunktion auszutesten. (?)

@Myprr Ja, ich denke es wäre sinnvoll wenn wir zuerst der Migration (Umsetzung) der Datenabank auf Prod anfangen.

**Comment by pnt-linii** [comment]
UPDATE:
- Registrierungslink funktioniert.
- Funktionen auf der Webseite für Admin und Professoren funktionieren (z.B. User anlegen, löschen, Userprogress resetten,...)

**Comment by JulianeOrt** [comment]
Gibt es beim Quiz auch eine Text-to-Speech Funktion? in der Admin Seite bei CRM/ICM kann man das verwalten, daher die Frage ob das beim Quiz auch notwendig ist?

**Comment by JulianeOrt** [comment]
<img width="492" alt="Screenshot 2019-11-19 at 12 53 20" src="https://user-images.githubusercontent.com/36894965/69144425-9274fa00-0acb-11ea-84ed-10fe34d2df33.png">

Zwischen "Lecturers" und "new lecturer" sollte eine neue Tabelle hinein gefügt werden, die vom Studiengangsleiter ausgefüllt wird und die Kurse und Vorlesungen mit den Dozenten per Drop-Down Menu ausgewählt werden soll.

**Comment by JulianeOrt** [comment]
<img width="875" alt="Screenshot 2019-11-19 at 13 04 01" src="https://user-images.githubusercontent.com/36894965/69145082-209db000-0acd-11ea-9404-b1197364290e.png">

Bei der Lecturer Page sollte das dort stehen und durch das Mapping vom Admin/Studiengangsleiter muss der Dozent nur auf enable bzw. end Quiz drücken müssen. Hintergrund wie beim Quiz

**Comment by JulianeOrt** [comment]
Für die eingeloggte Student Seite soll nur die verfügbaren Quizzes angezeigt werden und im Quiz brauchen wir noch einen Button für "zur Auswertung abgeben", der das Quiz dann schließt. Der student hat nur eine Möglichkeit, das Quiz im "Prüfungsmodus" zu machen.

**Comment by JulianeOrt** [comment]
![WhatsApp Image 2019-11-19 at 13 09 38](https://user-images.githubusercontent.com/36894965/69145480-fe586200-0acd-11ea-87d1-3b76dde09d89.jpeg)
so ist etwa die Übersicht dazu

**Comment by pnt-linii** [comment]
> Für die eingeloggte Student Seite soll nur die verfügbaren Quizzes angezeigt werden und im Quiz brauchen wir noch einen Button für "zur Auswertung abgeben", der das Quiz dann schließt. Der student hat nur eine Möglichkeit, das Quiz im "Prüfungsmodus" zu machen.

Heißt es, dass die Studenten sollen keine Quizzes zum "Selbslernen" haben?

**Comment by JulianeOrt** [comment]
> > Für die eingeloggte Student Seite soll nur die verfügbaren Quizzes angezeigt werden und im Quiz brauchen wir noch einen Button für "zur Auswertung abgeben", der das Quiz dann schließt. Der student hat nur eine Möglichkeit, das Quiz im "Prüfungsmodus" zu machen.
> 
> Heißt es, dass die Studenten sollen keine Quizzes zum "Selbslernen" haben?

Doch, die sollen weiterhin so bleiben wie sie sind und nicht verändert werden, dafür brauchen sich die Studenten auch nicht einloggen. Zusätzlich dazu sollten aber "prüfungs-quizzes" hinzugefügt werden, und in der eingeloggten Sicht sind dann nur die verfügbaren quizzes zur Prüfung angezeigt.

**Comment by DigitaleMedienDHBW** [comment]
Hallo zusammen,

man meldet sich nur zum Quiz an, um an einer Leistungsbewertung teilzunehmen; jeder kann beliebig das Quiz anonym spielen ohne sich anzumelden. Wenn für einen Kurs bestimmte Module des Quiz zur Leistungsbewertung freigeschaltet wurden, können Studierende dieses Kurses diese Module genau einmal angemeldet spielen, aber natürlich beliebig oft ohne Anmeldung. 

Mit freundlichen Grüßen 
Prof. Peter Mayr 


p.s.:
* Wir sollten auch für angemeldete Quiz-Nutzer wir nur das Ergebnis der Auswertung mit Zeitstempel, nicht aber seine Spielestände zwischendurch oder sonstige Zustandsdaten der Quiz-Teilnahme speichern; das wird sonst zu aufwändig.
* Wenn sich jemand anmelden will, für den kein Quiz-Modul zur Bewertung ansteht (weil er schon alle offenen Quizze gespielt hat oder es gar keine gibt), sollten wir ihn freundlich abwimmeln - "Derzeit steht kein (weiteres) Quiz-Modul zur Bewertung an." - und zurück auf die Quiz-Einstiegsseite für ein anonymes Quiz-Spiel leiten.

**Comment by pnt-linii** [comment]
> Für die eingeloggte Student Seite soll nur die verfügbaren Quizzes angezeigt werden und im Quiz brauchen wir noch einen Button für "zur Auswertung abgeben", der das Quiz dann schließt. Der student hat nur eine Möglichkeit, das Quiz im "Prüfungsmodus" zu machen.

@IsabellaMoos Die Studenten werden zur main.jsp unter brillianQUIZ/WebContent weitergeleitet. Ich habe die Seite 30% basierend auf login.jsp erstellt, bitte um weitere Anpassung.

**Comment by IsabellaMoos** [comment]
> main.jsp unter brillianQUIZ/WebContent/frontend weitergeleitet. Ich habe die Seite 30% basierend auf login.jsp erstellt, bitte um weitere Anpassung

@pnt-linii ich kümmere mich darum

**Comment by LauraBrinks** [comment]
![image](https://user-images.githubusercontent.com/56722107/69421017-39ea6a80-0d20-11ea-90d5-e6a8274bd001.png)
Wichtig für Project Owner, dass alle drei Arten von Fischen auch im Hintergrund vorkommen.

**Comment by JulianeOrt** [comment]
@pnt-linii wie ist hier der Status?

**Comment by pnt-linii** [comment]
> @pnt-linii wie ist hier der Status? Müssen wir das in Sprint 4 weiter bearbeiten?

Leider gibt es ein Problem beim Deployment der neuen Funktionen. Ich versuche das Problem so schnell wie möglich zu lösen. Aber es sieht so aus, dass wir in Sprint 4 weiter arbeiten müssen.

**Comment by Myprr** [comment]
> 
> 
> > > TODOs:
> > > ```
> > > * "Passwort Vergessen-Seite" und "Registrierung-Seite" bedürfen Anpassungen bezüglich Designs. Bitte jemand sich darum kümmere.
> > > 
> > > * Fix Registrationsbestätigungslink (Link in Bestätigungsemail führt zur 404)
> > > ```
> > 
> > 
> > @Lara-u erstelle bitte separate tasks für die oberen Aufgaben und verlinke mit diesem Task.
> > @pnt-linii Task in diesem Issue ist also die Konfiguration von Test auf Prod zu migrieren und mit unserer aktiver Mailfunktion auszutesten. (?)
> 
> @Myprr Ja, ich denke es wäre sinnvoll wenn wir zuerst der Migration (Umsetzung) der Datenbank auf Prod anfangen.

DB ist auf Prod umgesetzt.

**Comment by pnt-linii** [comment]
Update: Daten für Admin_Homepage und Professor_Homepage werden bereitgestellt und können in jsp Dateien über getAttribute-Methode aufgerufen werden.

**Comment by JulianeOrt** [comment]
@pnt-linii können oder konnten für die Admin und Dozentenseite schon diese Hintergründe eingepflegt werden wie hier https://github.com/imbitdhbwma/brillianplatform/issues/66#issuecomment-557490599 beschrieben?

**Comment by IsabellaMoos** [comment]
> @pnt-linii können oder konnten für die Admin und Dozentenseite schon diese Hintergründe eingepflegt werden wie hier [#66 (comment)](https://github.com/imbitdhbwma/brillianplatform/issues/66#issuecomment-557490599) beschrieben?

@JulianeOrt Ich kümmere mich darum. Ich passe auch weiter die Passwort-zurücksetzen/Registrieren-Seiten beim Quiz an, sobald das Issue mit dem Krakenkopf/Launchbar behoben ist

**Comment by pnt-linii** [comment]
Update: Backend der Mappping-Funktion für Admin wurde implementiert. @IsabellaMoos Bitte die Seite im Frontend so ergänzen wie gewünscht.

**Comment by IsabellaMoos** [comment]
> Update: Backend der Mappping-Funktion für Admin wurde implementiert. @IsabellaMoos Bitte die Seite im Frontend so ergänzen wie gewünscht.

@pnt-linii @JulianeOrt Ich habe das Design für die Admin- und Dozentenoberfläche angepasst (Siehe Screenshot). Das einzige, was noch fehlt, ist die Entfernung des Passwortfeldes beim Loginformular (Wunsch des Product Owner), aber ohne Passwort kann ich mich derzeit noch nicht einloggen. Fehlt abgesehen davon sonst noch was?


![image](https://user-images.githubusercontent.com/56718099/69951606-d5849380-14f5-11ea-9e7c-431a0cd8ffcd.png)

**Comment by pnt-linii** [comment]
UPDATE on Backend Studenten-Homepage :
Die Studenten können jetzt nur die Quizzes sehen, die für ihren Kurs vom Dozent freigeschaltet sind.
Für die Seite werden die Quizzes "Objekt" des Datentyps `<ArrayList<String>` bereitgestellt, die aber zusammen in einer anderen ArrayList "eingepackt" werden. Ein Quiz-Objekt ist in folgender Reihenfolge (`quizzes`.`subject_name`,`subject_header`, `intro_text`, `img_link`) definiert.
@IsabellaMoos Bitte das Design der Seite anpassen. 

![image](https://user-images.githubusercontent.com/56722146/69961340-f35cf300-150b-11ea-89e3-ebb6b76f55b4.png)
